src/item.h

```diff
@@ -493,6 +510,23 @@ class item : public JsonSerializer, public JsonDeserializer
  mtype*   corpse;
  std::vector<item> contents;
 
+protected:
+        // amount of storage used by bag
+        int storage_occupied;
+public:
+        // amount of storage available in bag
+        inline int storage_left() const
+        {
+            return (get_storage() - storage_occupied);
+        }
+        // amount of storage used in the bag
+        inline int storage_used() const
+        {
+            return (get_storage() - storage_left());
+        }
+        // whether the mysteries of a bag have been explored.
+        bool has_been_opened;
```

Comment:
A bool is insufficient for this, we need to know if a particular player has opened the bag.  Since there's no stable way for a player to refer to a bag, this should probably be a set of player identifiers.